kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -544,92 +536,4 @@ private Map<String, DomainMetadata> loadDomainMetadataMapFromLog(
       throw new UncheckedIOException("Could not close iterator", ex);
     }
   }
-
-  /**
-   * Encapsulates CRC-related functionality and state for the LogReplay. This includes caching CRC
-   * info and extracting snapshot hints from CRC files.
-   *
-   * <p>This class uses {@code maybeGetNewerSnapshotHintAndUpdateCache} to calculate a {@code
-   * SnapshotHint} and also exposes a {@code getLastSeenCrcInfo} method. Their relationship is:
-   *
-   * <ul>
-   *   <li>We want to find the latest {@code SnapshotHint} to use during log replay for Protocol and
-   *       Metadata loading
-   *   <li>If we are not provided a SnapshotHint for this version, or are provided a stale hint, we
-   *       will try to read the latest seen (by file listing) CRC file (if it exists). If so, we
-   *       read it, cache it, and create a newer hint.
-   *   <li>Then, when {@code getLastSeenCrcInfo} is called, we will either use the cached CRCInfo
-   *       that we have already read, parsed, and cached; or, if it was never cached (because the
-   *       hint was sufficiently new) we will read it, parse it, and cache it for the first time
-   * </ul>
-   */
-  private class CrcInfoContext {
-    private final Engine engine;
-    private Optional<CRCInfo> cachedLastSeenCrcInfo;
-
-    CrcInfoContext(Engine engine) {
-      this.engine = requireNonNull(engine);
-      this.cachedLastSeenCrcInfo = Optional.empty();
-    }
-
-    /** Returns the CRC info persisted in the logSegment's lastSeenChecksum File */
-    public Optional<CRCInfo> getLastSeenCrcInfo() {
-      if (!cachedLastSeenCrcInfo.isPresent()) {
```

Comment:
Bug! What if there is no CRC at version N? Then we will check the cache -- see that this is EMPTY -- and then try and load it! And then what happens if someone invokes this method again? Well, we will repeat the same load attempt!

This really just needed to be an `Optional<Optional<CRCInfo>>` or a `Lazy<Optional<CRC<Info>>`